syntax = "proto3";

package bookstore;

// Definição do serviço de Gestão de Pedidos
service OrderService {
    rpc PlaceOrder (OrderRequest) returns (OrderResponse);
    rpc ListOrders (UserRequest) returns (OrderListResponse);
    rpc GetOrder (OrderGetterRequest) returns (OrderGetterResponse);
}

// Mensagens utilizadas pelo serviço
message OrderRequest {
    string user_id = 1;
    string b_id = 2;
    int32 quantity = 3;
}

message OrderResponse {
    string order_id = 1;
}

message UserRequest {
    string user_id = 1;
}

message OrderSummary {
    string order_id = 1;
    string hour = 2;
}

message OrderListResponse {
    repeated OrderSummary orders = 1;
}

message OrderGetterRequest {
    string user_id = 1;
    string order_id = 2;
}

message OrderGetterResponse {
    string b_id = 1;
    int32 quantity = 2;
    string hour = 3;
}
